tests/nightly/gpu/test_gpt2.py

```diff
@@ -95,3 +102,76 @@ def test_nospecialtok(self):
         gpt2.observe({'text': 'My name is', 'episode_done': True})
         response = gpt2.act()
         assert response['text'] == " John. I'm a man of"
+
+
+BATCHSIZE = 4
+
+
+@testing_utils.skipUnlessGPU
+class TestDistributed(unittest.TestCase):
+    _base_config = {
+        'task': 'integration_tests:overfit',
+        'model': 'hugging_face/gpt2',
+        'gpt2_size': 'small',
+        'text_truncate': 16,
+        'label_truncate': 8,
+        'beam_min_length': 8,
+        'inference': 'beam',
+        'beam_size': 1,
+        'batchsize': BATCHSIZE,
+        'add_special_tokens': True,
+    }
+
+    def setUp(self):
+        print(f'[Setting up test {self._testMethodName}]')
+
+    def _forced_parse(self, parser, opt):
+        parser.set_params(**opt)
+        parser.set_params(log_every_n_sec=10)
+        popt = parser.parse_args([])
+        # in some rare cases, like for instance if the model class also
+        # overrides its default params, the params override will not
+        # be taken into account.
+        for k, v in opt.items():
+            popt[k] = v
+        return popt
+
+    def _distributed_train_model(self, opt):
+        with testing_utils.tempdir() as tmpdir:
+            if 'model_file' not in opt:
+                opt['model_file'] = os.path.join(tmpdir, 'model')
+            if 'dict_file' not in opt:
+                opt['dict_file'] = os.path.join(tmpdir, 'model.dict')
+
+            parser = mp_train.setup_args()
+            popt = self._forced_parse(parser, opt)
```

Comment:
please add a TODO over here and in the multiprocessing tests to kill forced_parse after dictionaries build correctly